Changes made to base

bluehexagons opened the pull request:
Hey rezecib; I happened to check forks for the food guide the other day and noticed that you'd been maintaining a fork. I pulled a majority of the changes and the features you added. If you'd like, you can pull those changes to your fork and be up to date with the main repo. I'm pretty impressed by how much you were able to do considering how monstrous the codebase is, it probably took a fair bit of effort to figure out where to even start.

bluehexagons committed:
Style changes, minor proofreading

bluehexagons committed:
First refinement pass

Changed DLC checking to use bitmasks

Renamed from modes to DLC since it's more likely to be used that way

Some more minor proofreading changes

Now saves selected DLC

bluehexagons committed:
Food List and Recipe List now update with DLC selection

Misc cleanup

bluehexagons committed:
Content/link updates

bluehexagons committed:
Remade UI for DLC selection

Minor cosmetic changes

bluehexagons committed:
Updated minified file

Will change gh-pages branch to point to minified, leave master as source

bluehexagons commented:
Also, if you're OK with it, I'd like to point Wiki links and such back to my repository. I didn't do it yet since I wanted to pass it by you first, considering you've been maintaining your fork for a while and I haven't had the time to do anything.

rezecib commented:
Haha, yeah, it was a bit disorienting at first. Making the food additions
was pretty easy, but setting up the game-mode switches took a fair bit of
poking around the code.

Sure, I was planning to suggest merging it back once Shipwrecked released,
as I can update in changes as they're made pretty quickly (I've been
keeping the Lua code under version control). Feel free to update the wiki
links and stuff too.

I also had some plans to do some other changes, like making the rot display
functional (so it can show that dogfish and swordfish spoil into spoiled
fish), and giving a running total of the tag values for the current
ingredients. I've gotten bogged down with my classes in grad school though,
so I don't know if I'll get to that any time soon. In any case, if I do get
around to it, I can suggest another merge at that point.

On Sun, Feb 21, 2016 at 3:54 PM, bluehexagons notifications@github.com
wrote:

> Also, if you're OK with it, I'd like to point Wiki links and such back to
> my repository. I didn't do it yet since I wanted to pass it by you first,
> considering you've been maintaining your fork for a while and I haven't had
> the time to do anything.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/rezecib/foodguide/pull/3#issuecomment-186913857.

bluehexagons commented:
A local git repo was how I tracked changes to the game when I maintained it, too. That's why everything was named differently in the food/recipes, and also why food types have double quotes around them instead of single quotes like the rest of the code, since I'd just copy/paste most things in.

Those are some good ideas. If both of us end up having random times to add features, we should probably open tickets or something and be semi-organized about it, though I'm pretty bogged down myself so I'm not sure how often I'll end up working on it beyond managing pull requests. I haven't even played Reign of Giants.

bluehexagons committed:
Miscellaneous bug fixes

Fixed several raw ingredients not being used in statistics analyzer

Fixed infinite loop on statistics analyzer with 1 ingredient

Fixed error printed to console when picker is cleared

bluehexagons committed:
Added recipe exclusion to statistics analyzer

bluehexagons committed:
Updated minified JavaScript file

bluehexagons committed:
Redid statistics analyzer's ingredient selection

Will now show raw, but use idealized stat values for comparison

This reduces duplicate food items, and also uses dry stats

Added idealized stats to other food tables

Changed sorting to allow related items to be grouped manually

Dead lobster is uncookable, likely not mistake

Lobsters are usually boiled alive IRL :(

Miscellaneous cleanup

bluehexagons committed:
Updated minified file

bluehexagons committed:
Removed drying from idealized health/hunger

Changed health/hunger/sanity to show (cooked values) instead of ideal

 and only if different from raw

Improved sorting, now allows manual tagging of wet/raw versions, dry sorts last

bluehexagons committed:
Food table health/hunger/sanity now shows difference when cooked (+/-), if any
